src/webob/acceptparse.py

```diff
@@ -1641,86 +927,54 @@ def __add__(self, other):
           including any media type parameters, and *extension_params* is a
           ``str`` of the extension parameters segment of the header element,
           starting with the first '``;``'
-        * an :class:`AcceptValidHeader`, :class:`AcceptNoHeader`, or
-          :class:`AcceptInvalidHeader` instance
+        * an :class:`Accept` instance
         * object of any other type that returns a value for ``__str__``
 
-        If `other` is a valid header value or an :class:`AcceptValidHeader`
-        instance, then a new :class:`AcceptValidHeader` instance with the valid
-        header value is returned.
-
-        If `other` is ``None``, an :class:`AcceptNoHeader` instance, an invalid
-        header value, or an :class:`AcceptInvalidHeader` instance, a new
-        :class:`AcceptNoHeader` instance is returned.
+        The rules for adding values to a header are that the values are
+        appended if valid, or discarded. If everything is discarded then an
+        instance representing a missing header is returned.
```

Comment:
This reads as a bit unclear - maybe could do with some rewording, or the explanations from the docstrings of the three `.__add__`s could be copied over and updated?